src/main/scala/chiseltest/internal/BackendInterface.scala

```diff
@@ -2,111 +2,93 @@
 
 package chiseltest.internal
 
-import chiseltest.Region
 import chisel3._
 
-import scala.collection.mutable
+import scala.math.BigInt
 
-trait AbstractTesterThread
+/** SimulatorInterface usages in ThreadedBackend. */
+trait BackendInterface { this: ThreadedBackend =>
 
-class TesterThreadList(protected val elts: Seq[AbstractTesterThread]) {
-  def toSeq(): Seq[AbstractTesterThread] = elts
-
-  def join() {
-    Context().backend.doJoin(elts, None)
-  }
-
-  def joinAndStep(clock: Clock) {
-    Context().backend.doJoin(elts, Some(clock))
-  }
-
-  def ++(others: TesterThreadList): TesterThreadList = {
-    new TesterThreadList(elts ++ others.elts)
-  }
-
-  val fork: ForkBuilder = new ForkBuilder(None, None, elts)
-}
-
-/** Common interface definition for tester backends. Internal API.
-  */
-trait BackendInterface {
+  def getClock(clk: Clock): Boolean =
+    simulatorInterface.peek(dataNames(clk)) match {
+      case Some(x) if x == BigInt(1) => true
+      case _                         => false
```

Comment:
Should this really be the result for the None case? Unless it makes sense to fail silently here (I think this usually should never be the case), I think an exception might make more sense? Also see above comment in the definition of peek.